tests/deepsparse/pipelines/test_dynamic_batch_pipeline.py

```diff
@@ -0,0 +1,146 @@
+    elif isinstance(expected, dict):
+        assert list(expected.keys()).sort() == list(actual.keys()).sort()
+        for key, exp_value in expected.items():
+            assert compare(exp_value, actual[key])
+    else:
+        assert expected == actual
+    return True
+
+
+@pytest.mark.parametrize("task, batch_size", tasks_batch_size_pairs, scope="class")
+class TestDynamicBatchPipeline:
+    @pytest.fixture()
+    def dynamic_batch_pipeline(self, task, executor):
+        """
+        An auto-delete fixture to yield a Dynamic Batch Pipeline, this Pipeline
+        is capable of executing different batch sizes.
+        """
+        yield Pipeline.create(
+            task=task,
+            batch_size=None,
+            executor=executor,
+        )
+
+    @pytest.fixture()
+    def static_batch_pipeline(self, task, executor, batch_size):
+        """
+        An auto-delete fixture to yield a Static Batch Pipeline
+        """
+        assert batch_size is not None
+        yield Pipeline.create(
+            task=task,
+            batch_size=batch_size,
+            executor=executor,
+        )
+
+    @pytest.fixture()
+    def inputs(self, task, batch_size):
+        """
+        An auto-delete fixture to get task inputs
+        """
+        yield create_test_inputs(task=task, batch_size=batch_size)
+
+    @pytest.fixture()
+    def dynamic_batch_outputs(self, inputs, dynamic_batch_pipeline):
+        """
+        An auto-delete fixture to yield output from dynamic batch pipline
+        """
+        yield dynamic_batch_pipeline(**inputs)
+
+    @pytest.fixture()
+    def static_batch_outputs(self, inputs, static_batch_pipeline):
+        """
+        An auto-delete fixture to yield output from dynamic batch pipline
+        """
+        results = static_batch_pipeline(**inputs)
+        if isinstance(results, concurrent.futures.Future):
+            yield results.result()
+        else:
+            yield results
+
+    def test_pipeline_creation(self, batch_size, dynamic_batch_pipeline):
+        # Will fail if fixture request fails
+        pass
+
+    def test_execution_and_output(
+        self,
+        dynamic_batch_outputs,
+        dynamic_batch_pipeline,
+    ):
+        assert dynamic_batch_outputs
+        assert not isinstance(dynamic_batch_outputs, concurrent.futures.Future), (
+            "Expected dynamic batch pipeline to be blocking but got"
+            "got a concurrent.futures.Future object instead"
+        )
+        assert type(dynamic_batch_outputs) == dynamic_batch_pipeline.output_schema
+
+    def test_order_retention_against_static_batch(
+        self,
+        static_batch_outputs,
+        dynamic_batch_outputs,
+    ):
+        expected_dict = static_batch_outputs.dict()
+        actual_dict = dynamic_batch_outputs.dict()
+
+        # Check that order is maintained
+        assert static_batch_outputs == dynamic_batch_outputs or compare(
```

Comment:
if all inputs in the test input are the same, wouldn't this test pass even if the order is mutated?